### PR TITLE
[improve][doc] Add doc for improving source performance

### PR DESCRIPTION
**Pull request by danpi**
### Motivation

#487
 
### Modifications

- Add a description of the `batchSizeOfOnceReceive` and the`numberOfConsumers` configuration items.

## Documentation

- [x] `doc-required`

### CODE REVIEW

**Comment by danpi** [comment]
@Huanli-Meng PTAL, thanks.

File: docs/source_performance_test.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+### Step2:  Add the numberOfConsumers
+In this scene, we set the `numberOfConsumers`=10 and `batchSizeOfOnceReceive`=1. The figure below shows that Tps increases linearly (The Tps = 400/s = initialTPS*10).
+<img width="921" alt="image" src="https://user-images.githubusercontent.com/6015134/199722036-ed9783dd-c376-4356-99e5-4517d3a34b24.png">
+
+### Step3:  Change both of them
+Try to set the `numberOfConsumers`=10 and `batchSizeOfOnceReceive`=10. We want a 100-fold increase in Tps. The results are shown below (Tps=4000/s=100*`initialTPS`)
+![image](https://user-images.githubusercontent.com/6015134/200255470-7eb20144-0b05-4604-a8fc-db2c880dc091.png)
+
+
+### Step4: Starting multiple source Pods
```
Comment (Huanli-Meng) [inline docs/source_performance_test.md:37]:
```suggestion
### Step4: Start multiple source Pods
```

File: docs/source_performance_test.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+The resources of the test environment limited us. We set `parallelism`=2 and keep the `numberOfConsumers`=30 and `batchSizeOfOnceReceive`=10. The Tps will eventually reach about 20000/s.
```
Comment (Huanli-Meng) [inline docs/source_performance_test.md:38]:
```suggestion
The resources of the test environment are limited to us. Set `parallelism` to 2, set `numberOfConsumers` to 30, and set `batchSizeOfOnceReceive` to 10. The Tps eventually reach 20000/s.
```

File: docs/source_performance_test.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+![image](https://user-images.githubusercontent.com/6015134/200277829-23f9a2dc-e717-44d1-8169-18cfe1749c52.png)
+
+You can see that the Tps of a single pod is around 10_000/s.
+<img width="1709" alt="image" src="https://user-images.githubusercontent.com/6015134/200277460-54d7c8f6-b0ec-40eb-aa62-30d261b8e2bf.png">
+
+The CPU was exhausted due to resource limitations in the test environment, so the relationship between Tps and consumer thread count is not completely linear.
+![image](https://user-images.githubusercontent.com/6015134/200277562-680be200-3d69-4596-ac9e-9a20ba15cdc0.png)
+
+## Summarize
+From the above tests, we can conclude that the Tps can be significantly improved. There are some suggestions for using it.
```
Comment (Huanli-Meng) [inline docs/source_performance_test.md:48]:
```suggestion
From the above tests, it is concluded that the Tps can be significantly improved. Here are some suggestions for using it.
```

File: docs/source_performance_test.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+Desired result:
+1. When we increase the `batchSizeOfOnceReceive`=10, the throughput will be ten times what it was before.
+2. If we change the `numberOfConsumers`=10, the throughput increases linearly.
+3. If we increase them both, the throughput of a single node can reach 10000/s.
+
+## Operating environment
+The test environment is the same as before #484.
+## Test Step
+### Step1: Increase the batchSizeOfOnceReceive
```
Comment (Huanli-Meng) [inline docs/source_performance_test.md:19]: It seems that Step 1- step 4 are different scenarios. There is no specific order. If I understand it correctly, I think words like "Senario 1" is more appropriate.

File: docs/sqs-source.md [hunk 5]
```diff
@@ -51,6 +51,13 @@ Before using the SQS source connector, you need to configure it. Below are the p
...
 | `queueName`                | String | true     | " " (empty string) | Name of the SQS queue that messages should be read from or written to.                                                                                                                                                                                                                                                                                       |
+| `batchSizeOfOnceReceive`   | int    | false    | 1                  | For SQS source. The maximum number of messages pull from SQS at one time. Default=1 and the value range is [1,10].                                                                                                                                                                                                                                                         |
```
Comment (Huanli-Meng) [inline docs/sqs-source.md:54]:
```suggestion
| `batchSizeOfOnceReceive`   | int    | false    | 1                  | The maximum number of messages that are pulled from SQS at one time. By default, it is set to 1. The value ranges from 1 to 10.                                                                                                                                                                                                                                                         |
```

File: docs/sqs-source.md [hunk 6]
```diff
@@ -51,6 +51,13 @@ Before using the SQS source connector, you need to configure it. Below are the p
...
+| `numberOfConsumers`        | int    | false    | 1                  | For SQS source. The expected numbers of consumers. You can scale message consumers horizontally to achieve high throughput. Default=1 and the value range is [1,50].                                                                                                                                                                                                       |
```
Comment (Huanli-Meng) [inline docs/sqs-source.md:55]:
```suggestion
| `numberOfConsumers`        | int    | false    | 1                  | The expected numbers of consumers. You can scale consumers horizontally to achieve high throughput. By default, it is set to 1. The value ranges from 1 to 50.                                                                                                                                                                                                       |
```

File: docs/sqs-source.md [hunk 7]
```diff
@@ -51,6 +51,13 @@ Before using the SQS source connector, you need to configure it. Below are the p
...
+
+> **Note**  
+> You have to focus the version if you use `numberOfConsumers` and `numberOfConsumers` to improve the SQS source performance.
```
Comment (Huanli-Meng) [inline docs/sqs-source.md:58]:
```suggestion
> The `batchSizeOfOnceReceive ` and `numberOfConsumers` options are available for SQS source 2.8.4.3+, 2.9.4.1+, and 2.10.1.13+. For details about how to test AWS SQS source performance, see [Performance Test on AWS SQS Source Connector](/docs/source_performance_test.md).
```

File: docs/sqs-source.md [hunk 8]
```diff
@@ -51,6 +51,13 @@ Before using the SQS source connector, you need to configure it. Below are the p
...
+| `batchSizeOfOnceReceive`   | int    | false    | 1                  | For SQS source. The maximum number of messages pull from SQS at one time. Default=1 and the value range is [1,10].                                                                                                                                                                                                                                                         |
+| `numberOfConsumers`        | int    | false    | 1                  | For SQS source. The expected numbers of consumers. You can scale message consumers horizontally to achieve high throughput. Default=1 and the value range is [1,50].                                                                                                                                                                                                       |
+
+> **Note**  
+> You have to focus the version if you use `numberOfConsumers` and `numberOfConsumers` to improve the SQS source performance.  
+> The versions that support these features are: 2.8.4.3+,2.9.4.1+,2.10.1.13+.
```
Comment (Huanli-Meng) [inline docs/sqs-source.md:59]:
```suggestion

```

File: docs/sqs-source.md [hunk 9]
```diff
@@ -51,6 +51,13 @@ Before using the SQS source connector, you need to configure it. Below are the p
...
+| `batchSizeOfOnceReceive`   | int    | false    | 1                  | For SQS source. The maximum number of messages pull from SQS at one time. Default=1 and the value range is [1,10].                                                                                                                                                                                                                                                         |
+| `numberOfConsumers`        | int    | false    | 1                  | For SQS source. The expected numbers of consumers. You can scale message consumers horizontally to achieve high throughput. Default=1 and the value range is [1,50].                                                                                                                                                                                                       |
+
+> **Note**  
+> You have to focus the version if you use `numberOfConsumers` and `numberOfConsumers` to improve the SQS source performance.  
+> The versions that support these features are: 2.8.4.3+,2.9.4.1+,2.10.1.13+.  
+> Here is the performance test report: https://github.com/streamnative/pulsar-io-sqs/issues/486.
```
Comment (Huanli-Meng) [inline docs/sqs-source.md:60]:
```suggestion

```

**Review by Huanli-Meng — COMMENTED** [review]
Leave my comments. PTAL

**Comment by Huanli-Meng** [comment]
Remove the `doc-required` label as the `doc-required` label is used to indicate a code PR that requires updating/adding docs.

File: docs/source_performance_test.md (hunk 4 above)
Comment (danpi) [reply docs/source_performance_test.md:22]: Good suggestion, I will revise it.

**Comment by danpi** [comment]
@Huanli-Meng Apply your suggestions. PTAL, thanks.

File: docs/source_performance_test.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+2022-11-03T09:41:25,198+0000 [pulsar-timer-5-1] INFO  org.apache.pulsar.client.impl.ProducerStatsRecorderImpl - [test-queue-pulsar] [pulsar-108-31] Pending messages: 0 --- Publish throughput: 376.65 msg/s --- 0.37 Mbit/s --- Latency: med: 11.000 ms - 95pct: 15.000 ms - 99pct: 16.000 ms - 99.9pct: 39.000 ms - max: 76.000 ms --- BatchSize: med: 10.000 - 95pct: 10.000 - 99pct: 10.000 - 99.9pct: 10.000 - max: 10.000 --- MsgSize: med: 1280.000 bytes - 95pct: 1280.000 bytes - 99pct: 1280.000 bytes - 99.9pct: 1280.000 bytes - max: 1280.000 bytes --- Ack received rate: 376.65 ack/s --- Failed messages: 0 --- Pending messages: 0
+```
+
+### Senario 2:  Add the numberOfConsumers
```
Comment (Huanli-Meng) [inline docs/source_performance_test.md:31]:
```suggestion
### Scenario 2: Add the numberOfConsumers
```

File: docs/source_performance_test.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+Desired result:
+1. When we set the `batchSizeOfOnceReceive` to 10, the throughput is ten times greater than before.
+2. When we set the `numberOfConsumers` to 10, the throughput increases linearly.
+3. When we set both, the throughput of a single node can reach 10000/s.
+
+## Operating environment
+The test environment is the same as before #484.
+## Test Senarios
+### Senario 1: Increase the batchSizeOfOnceReceive
```
Comment (Huanli-Meng) [inline docs/source_performance_test.md:22]:
```suggestion
### Scenario 1: Increase the batchSizeOfOnceReceive
```

File: docs/source_performance_test.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+2022-11-03T09:41:25,198+0000 [pulsar-timer-5-1] INFO  org.apache.pulsar.client.impl.ProducerStatsRecorderImpl - [test-queue-pulsar] [pulsar-108-31] Pending messages: 0 --- Publish throughput: 376.65 msg/s --- 0.37 Mbit/s --- Latency: med: 11.000 ms - 95pct: 15.000 ms - 99pct: 16.000 ms - 99.9pct: 39.000 ms - max: 76.000 ms --- BatchSize: med: 10.000 - 95pct: 10.000 - 99pct: 10.000 - 99.9pct: 10.000 - max: 10.000 --- MsgSize: med: 1280.000 bytes - 95pct: 1280.000 bytes - 99pct: 1280.000 bytes - 99.9pct: 1280.000 bytes - max: 1280.000 bytes --- Ack received rate: 376.65 ack/s --- Failed messages: 0 --- Pending messages: 0
+```
+
+### Senario 2:  Add the numberOfConsumers
+In this scene, set the `numberOfConsumers` to 10 and `batchSizeOfOnceReceive` to1. The figure below shows that Tps increases linearly (The Tps = 400/s = initialTPS*10).
+<img width="921" alt="image" src="https://user-images.githubusercontent.com/6015134/199722036-ed9783dd-c376-4356-99e5-4517d3a34b24.png">
+
+### Senario 3:  Change both of them
```
Comment (Huanli-Meng) [inline docs/source_performance_test.md:35]:
```suggestion
### Scenario 3: Change both of them
```

File: docs/source_performance_test.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+Set `numberOfConsumers` to 10 and `batchSizeOfOnceReceive` to 10. We expect a 100-fold increase in Tps. The results are shown below (Tps=4000/s=100*`initialTPS`)
+![image](https://user-images.githubusercontent.com/6015134/200255470-7eb20144-0b05-4604-a8fc-db2c880dc091.png)
+
+
+### Senario 4: Start multiple source Pods
```
Comment (Huanli-Meng) [inline docs/source_performance_test.md:40]:
```suggestion
### Scenario 4: Start multiple source Pods
```

**Review by Huanli-Meng — APPROVED** [review]
leave some minor comments.

**Review by Huanli-Meng — APPROVED** [review]

**Review by freeznet — APPROVED** [review]

**Pull request merged**